### PR TITLE
Move transaction retry timeseries to TOML

### PR DESCRIPTION
**Pull request by bnaecker**

### CODE REVIEW

File: oximeter/oximeter/schema/database-transaction.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+format_version = 1
```
Comment (smklein) [inline oximeter/oximeter/schema/database-transaction.toml:1]: Is the primary motivation for this shift "better detection/management of schema changes?"

File: oximeter/oximeter/schema/database-transaction.toml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+authz_scope = "fleet"
+versions = [
+    { version = 1, fields = [ "name" ] }
+]
+
+[[metrics]]
+name = "retry_data"
+description = "Information about a retried transaction"
+units = "bytes"
+datum_type = "f64"
```
Comment (smklein) [inline oximeter/oximeter/schema/database-transaction.toml:15]: `RetryData` previously included a `latency` datum (f64) which was "seconds".

I'm not sure I'm seeing where that's represented here. I do see the `attempt` number below on line 25ish, but is it  right to say that the units are "byte" here?

File: oximeter/oximeter/schema/database-transaction.toml (hunk 1 above)
Comment (bnaecker) [reply oximeter/oximeter/schema/database-transaction.toml:1]: Yeah, that's the main reason. There are a bunch of pieces to this, tracked under #5941, and this is part of the "move all extant timeseries to TOML" item. Another good reason is that we can attach metadata, like the descriptions, that go in the DB but not each sample that `oximeter` collects. We can also generate docs for each timeseries, which is nice.

File: oximeter/oximeter/schema/database-transaction.toml (hunk 2 above)
Comment (bnaecker) [reply oximeter/oximeter/schema/database-transaction.toml:15]: Ah, the unit is wrong, thanks for catching that. The `latency` field is now `datum`. I did not add the ability to rename the actual datum field in the TOML definitions, which is what we'd done here with the `#[datum]` attribute on the `latency` field.

**Review by smklein — APPROVED** [review]
LGTM modulo getting the right units for latency!

File: oximeter/oximeter/schema/database-transaction.toml (hunk 2 above)
Comment (bnaecker) [reply oximeter/oximeter/schema/database-transaction.toml:15]: I'm going to queue this behind #6006, which adds the `Unit::Seconds` variant that this needs. I'll rebase and merge after that integrates into `main`.

Comment (bnaecker) [reply oximeter/oximeter/schema/database-transaction.toml]: Picked up the seconds variant in `1b77125f5`. I also found and fixed a bug in how extra traits are derived for the generated types. That is enough code that it deserves a second look @smklein. Thanks!

**Review by smklein — APPROVED** [review]

**Pull request merged**